src/transactionIntentsFactories/delegationTransactionIntentsFactory.ts

```diff
@@ -0,0 +1,344 @@
+import { IAddress } from "../interface";
+import { BigNumber } from "bignumber.js";
+import { numberToPaddedHex, byteArrayToHex, utf8ToHex } from "../utils.codec";
+import { TransactionIntent } from "../transactionIntent";
+import { TransactionIntentBuilder } from "./transactionIntentBuilder";
+import { Address } from "../address";
+import { DELEGATION_MANAGER_SC_ADDRESS } from "../constants";
+import { Err } from "../errors";
+
+interface Config {
+    chainID: string;
+    minGasLimit: BigNumber.Value;
+    gasLimitPerByte: BigNumber.Value;
+    gasLimitStake: BigNumber.Value;
+    gasLimitUnstake: BigNumber.Value;
+    gasLimitUnbond: BigNumber.Value;
+    gasLimitCreateDelegationContract: BigNumber.Value;
+    gasLimitDelegationOperations: BigNumber.Value;
+    additionalGasLimitPerValidatorNode: BigNumber.Value;
+    additionalGasLimitForDelegationOperations: BigNumber.Value;
+}
+
+interface ValidatorPublicKey {
```

Comment:
This is an internal interface, but even so, by convention, we can prefix it with `I`. Indeed, this is not the general practice (e.g. in other projects), but in `sdk-js-*` we did so.

Reply:
Renamed to `IValidatorPublicKey`